dowhy/causal_discovery.py

```diff
@@ -0,0 +1,32 @@
+class GraphLearner:
+	"""Base class for an causal discovery methods.
+
+	Subclasses implement different discovery methods. All discovery methods are in the package "dowhy.causal_discoverers"
+
+	"""
+
+	def __init__(self, data, library_class, *args, **kwargs):
```

Comment:
you need to store library class inside a self variable. also library_name is a better name, since it is just a string, right?

Reply:
It's not a string but an object of the class. I basically just pass the object name. For example, if Func() is the object, I pass Func as a variable var so that later, var() == Func(). So I named it library class. Library name does not seem right because I do not pass the name. Do you have some other suggestion for the name?

Reply:
I feel that it is safer to create the object inside graph_learner.py. That way the code for causal_model is light wrapper, and has less chance of raising exceptions. 
Similarly, the main task of importing the library and instantiating the class, should be inside graph_learner.

In that case, you can pass the string, and call it ``full_method_name''